doc/source/guide/in_depth/numpy_guide.rst

```diff
@@ -0,0 +1,124 @@
+<http://docs.scipy.org/doc/numpy/reference/ufuncs.html>`_) this is supported both as member functions and using the NumPy functions. For example, using NumPy::
+
+   >>> r3 = odl.Rn(3)
+   >>> x = r3.element([1, 2, 3])
+   >>> np.negative(x)
+   Rn(3).element([-1.0, -2.0, -3.0])
+
+This method of using NumPy always uses the NumPy implementation, which can occur overhead in case the data is not stored in a CPU space. To always enable optimized code, users can call the member `NtuplesBaseVector.ufunc`::
+
+   >>> x.ufunc.negative()
+   Rn(3).element([-1.0, -2.0, -3.0])
+
+For other arbitrary functions, ODL vectors are generally accepted as input, but the output is often of `numpy.ndarray` type::
+
+   >>> np.convolve(x, x, mode='same')
+   array([  4.,  10.,  12.])
+
+
+NumPy functions as Operators
+----------------------------
+To solve the above issue, it is often useful to write an `Operator` wrapping NumPy functions, thus allowing full access to the ODL ecosystem. To wrap the convolution operation, a user could write a new class::
+
+   class MyConvolution(odl.Operator):
+       """Operator for convolving with given vector"""
+
+       def __init__(self, vector):
+           """Create a Convolution by vector"""
+           self.vector = vector
+
+           # Initialize operator superclass. This is a operator from the space
+           # of ``vector`` to itself which is linear. Setting linear=True allows
+           # some optimizations and is required by some methods such as linear
+           # solvers
+           odl.Operator.__init__(self, domain=vector.space, range=vector.space,
+                                 linear=True)
+
+       def _call(self, x):
+           # The output of an Operator is automatically cast to an ODL vector
+           return np.convolve(x, self.vector, mode='same')
+
+This could then be called as an ODL Operator::
+
+   >>> op = MyConvolution(x)
+   >>> op(x)
+   Rn(3).element([4.0, 10.0, 12.])
+
+Since this is an ODL Operator, it can be used with any of the ODL functionalities such as multiplication with scalar, composition, etc::
+
+   >>> scaled_op = 2 * op  # scale by scalar
+   >>> scaled_op(x)
+   Rn(3).element([8.0, 20.0, 24.])
+   >>> y = r3.element([1, 1, 1])
+   >>> inner_product_op = odl.InnerProductOperator(y)
+   >>> composed_op = inner_product_op * op  # create composition with inner product with vector [1, 1, 1]
+   >>> composed_op(x)
+   26.0
+
+Using ODL with SciPy linear solvers
+-----------------------------------
+SciPy includes `a series of very competent solvers
+<http://docs.scipy.org/doc/scipy/reference/sparse.linalg.html>`_ that may be useful in solving some linear problems. If you have invested some effort into writing an ODL operator, or perhaps wish to use a pre-existing operator then the function `as_scipy_operator` can allow you to use these together, here is a simple example of solving Poisson's equation equation on an interval (:math:`- \Delta x = rhs`)::
```

Comment:
Suggestion:

    "[...], then the member function `as_scipy_operator` creates a Python object that can be used
    in SciPy's linear solvers. Here is a [...]"